### PR TITLE
Add gdbm/0.18.1 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **gdbm/1.18.1**

Port of bincrafters recipe

Depends on #1195

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@SSE4 @uilianries @danimtb 
image

**Comment by madebr** [comment]
@uilianries @SSE4  @danimtb @jgsogo 
Please review!

File: recipes/gdbm/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class GdbmConan(ConanFile):
+    name = "gdbm"
+    version = "1.18.1"
```
Comment (uilianries) [inline recipes/gdbm/all/conanfile.py:8]:
```suggestion
```

File: recipes/gdbm/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        if self.options.with_libiconv:
+            self.requires("libiconv/1.16")
+        if self.options.with_readline:
+            raise ConanInvalidConfiguration("readline is not (yet) available on CCI")
+            self.requires("readline/x.y")
```
Comment (uilianries) [inline recipes/gdbm/all/conanfile.py:55]:
```suggestion
            # TODO - Add readline package when available
```

**Comment by uilianries** [comment]
I've removed "locked" because now we have `flex` available as Conan package in CCI.

**Pull request merged**